src/App.js

```diff
@@ -1,5 +1,145 @@
+import { MissionUtils } from "@woowacourse/mission-utils";
+
+class CarNameParser {
+  constructor(namesString) {
+    CarNameError.validateInputString(namesString);
+    const parsedNames = this.parseNames(namesString);
+    CarNameError.validateEmptyName(parsedNames);
+    this.names = parsedNames;
+    this.validateNames();
+  }
+
+  parseNames(input) {
+    return input.split(',')
+      .map(name => name.trim())
+      .filter(name => name.length > 0);
+  }
+
+  validateNames() {
+    this.names.forEach(name => CarNameError.validateNameLength(name));
+    CarNameError.validateDuplicateNames(this.names);
+  }
+
+  getNames() {
+    return this.names;
+  }
+}
+
+class CarNameError {
+  static validateDuplicateNames(names) {
+    const uniqueNames = new Set(names);
+    if (uniqueNames.size !== names.length) {
+      throw new Error("[ERROR] 이름이 중복되었습니다.");
+    }
+  }
+
+  static validateNameLength(name) {
+    const MAX_NAME_LENGTH = 5;
+    if (name.length > MAX_NAME_LENGTH) {
+      throw new Error("[ERROR] 이름은 5자를 초과할 수 없습니다.");
+    }
+  }
+
+  static validateEmptyName(names) {
+    if (names.length === 0) {
+      throw new Error("[ERROR] 최소 하나의 이름이 필요합니다.");
+    }
+  }
+
+  static validateInputString(input) {
+    if (!input || typeof input !== 'string') {
+      throw new Error("[ERROR] 유효하지 않은 입력입니다.");
+    }
+    
+    if (input.trim() === '') {
+      throw new Error("[ERROR] 최소 하나의 이름이 필요합니다.");
+    }
+  }
+  
+}
+
+class TimeStringToNumber {
+  static parse(times) {
+    TimeStringToNumber.isPositiveInteger(times);
+    return Number(times);
+  }
+
+  static isPositiveInteger(times) {
+    const POSITIVE_INTEGER_REGEX = /^\d+$/;
+    if (POSITIVE_INTEGER_REGEX.test(times) == false) {
+      throw new Error("[ERROR]횟수는 양의 정수이어야 합니다.");
+    }
+  }
+}
+
+class CarRace {
+  constructor(playerNames) {
+    this.playerNames = playerNames;
+    this.countArray = new Array(playerNames.length).fill(0);
+  }
+
+  race(time) {
+    MissionUtils.Console.print("실행 결과\n");
+    for (let i = 0; i < time; i++) {
+      this.runRound();
+      MissionUtils.Console.print('\n');
+    }
```

Comment:
우아한테크코스의 스타일 규칙인 Airbnb는 를 지양하고 있어서, 반복문을 이런식으로 바꿔보시는 것은 어떨지 추천드리고 싶습니다.

```suggestion
const times = new Array(time);
times.forEach((time) => {
  this.runRound();
  MissionUtils.Console.print('\n');
});
```

[관련 링크]
https://github.com/airbnb/javascript?tab=readme-ov-file#iterators-and-generators